frontend/src/components/GrantAccessForm.vue

```diff
@@ -0,0 +1,183 @@
+
+        const apiService = new ApiService();
+
+        const defaultFormData = {
+            domain: domainOptions.BCEID,
+            userId: null,
+            forestClientNumber: null,
+            role: null as unknown as ApplicationRoleResponse
+        };
+
+        const formData = reactive({ ...defaultFormData });
+        const rules = {
+            domain: {},
+            userId: { required, minLength: minLength(3), maxLength: maxLength(20) },
+            role: { required },
+            forestClientNumber: {},
+        };
+
+        const initialState = {
+            name: "",
+            lastName: "",
+            email: ""
+        };
+
+        function resetForm() {
+            Object.assign(formData, defaultFormData);
+        };
+
+        const v$ = useVuelidate(rules, formData);
+
+        return { formData, v$, domainOptions, apiService, FOREST_CLIENT_INPUT_MAX_LENGTH, defaultFormData, resetForm }
+    },
+    data() {
+        return { applicationRoleOptions: ref<ApplicationRoleResponse[]>([]), buttonDisabled: true }
+    },
+    computed: {
+        onBlur() {
+            this.v$.$validate();
+            if (this.v$.$errors) this.buttonDisabled = true;
+            else this.buttonDisabled = false;
+        }
+    },
+    methods: {
+        onlyDigit(evt: KeyboardEvent) {
+            if (isNaN(parseInt(evt.key))) {
+                evt.preventDefault();
+            }
+        },
+
+        async grantAccess() {
+            this.v$.$validate();
+            if (this.v$.$error) return;
+            const toast = useToast();
+            const grantAccessRequest = this.toRequest(this.formData);
+            try {
+                await this.apiService.grantUserRole(grantAccessRequest);
+                toast.success(`User "${grantAccessRequest.user_name}" is granted with "${this.formData?.role.role_name}" access.`);
+                this.resetForm();
+                this.v$.$reset();
+            }
+            catch (err: any) {
+                useToast().error(`Grant Access failed due to an error. Please try again.If the error persists then contact support.\nMessage: ${err.response.data?.detail}`);
+                console.error("err: ", err);
+            }
+        },
+
+        toRequest(formData: any) {
+            const request = {
+                user_name: formData.userId,
+                user_type_code: formData.domain,
+                role_id: formData.role.role_id,
+                ...(formData.forestClientNumber ?
+                    { forest_client_number: formData.forestClientNumber.padStart(this.FOREST_CLIENT_INPUT_MAX_LENGTH, '0') }
+                    : {})
+            } as GrantUserRoleRequest;
+
+            return request;
+        }
+    },
+    async mounted() {
+        this.applicationRoleOptions = await this.apiService.getApplicationRoles(
+            selectedApplication?.value?.application_id
+        ) as ApplicationRoleResponse[];
+    }
+}
+
+</script>
+
+<style scoped>
+
```

Comment:
Empty style section is being flagged in SonarCloud. Remove section if not needed.

Reply:
My mistake, it will be fixed in the next commit.